### ISSUE TITLE
[NEXT-1124] URL properties of `metadata` are broken when resolved from parent

### ISSUE THREAD

**Issue by rafaelalmeidatk** [vercel/next.js#49406]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #22 SMP Tue Jan 10 18:39:00 UTC 2023
Binaries:
  Node: 16.17.0
  npm: 8.15.0
  Yarn: 1.22.19
  pnpm: 7.1.0
Relevant packages:
  next: 13.4.1-canary.2
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Metadata (metadata, generateMetadata, next/head, head.js)

### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/fervent-snowflake-me2gqj?selection=%5B%7B%22endColumn%22%3A41%2C%22endLineNumber%22%3A3%2C%22startColumn%22%3A5%2C%22startLineNumber%22%3A3%7D%5D&file=%2Fapp%2Flayout.tsx

### To Reproduce

1. Open the sandbox link
2. Click the link to go to the `/test` page
3. The page crashes

### Describe the Bug

When adding a property related to `URL` like `openGraph.url` or `openGraph.images` in `app/layout.tsx`, the field becomes invalid when retrieving it in nested segments using `await parent`, e.g.:

```tsx
export async function generateMetadata(
  _props: unknown,
  parent?: ResolvingMetadata
): Promise<Metadata> {
  const parentMetadata = await parent;
  // this should work
  return parentMetadata;
}
```

This is problematic when trying to merge the parent metadata with a new object, since unless the new object overrides these exact fields, they will be invalid thus resulting in the crash

### Expected Behavior

The return of `await parent` should always be valid

<sub>[NEXT-1124](https://linear.app/vercel/issue/NEXT-1124/url-properties-of-metadata-are-broken-when-resolved-from-parent)</sub>

**Comment by mateuszaliyev** [comment]
Can confirm, having the same issue.

**Comment by frankyw** [comment]
Not sure if this is related, but also have problems with certain metadata URLs, related to metadatabase (which is not configured).

`  return {
    alternates: {
      canonical: `/blog/`,
    },
    openGraph: {
      url: `/blog/`,
      images: [
        {
          url: '/og-main.png',
`

- canonical - does not include any hostname: `<link rel="canonical" href="/blog/">`
- og:url - includes "http://localhost:3000/" when deployed to Vercel: `<meta property="og:url" content="http://localhost:3000/blog/">`
- og:image - works fine when deployed to Vercel `<meta property="og:image" content="https://xxxxx.vercel.app/og-main.png">`

**Comment by Diogomartf** [comment]
I think I have a similar  issue as above (@frankyw).

By adding the `twitter-image.png` and `opengraph-image.png` to the `app` root, I get the `http://localhost:3000` on prod. 

is that a bug?

 couldn't find how to set the base url on the docs

**Comment by danilo900** [comment]
Same issue here, gettin "localhost:3000" on prod when using ** openGraph** image url pointing to **/image.jpg**

**Comment by Diogomartf** [comment]
is there some updates on this issue?

**Comment by huozhi** [comment]
Sorry for the late visit, this should be fixed in #49343 alreay. Upgrade will resolve the issue 🙏

**Comment by Diogomartf** [comment]
In which version is fixed? 

Tried updating to canary and set `metadataBase` on metadata (on root layout.js) and still got localhost.

**you can see the PR here**
https://github.com/SwaprHQ/stackly-ui/pull/11

**Comment by Diogomartf** [comment]
Thanks for your work 🙌

Seems to be fixed on `next@canary` but not on `next@13.4.7` (latest release).

Probably will be fixed on `next@13.4.8`.
